Console/Command/BatchingOptimizeCommand.php

```diff
@@ -0,0 +1,366 @@
+    {
+        if (count($storeIds)) {
+            foreach ($storeIds as $storeId) {
+                $this->scanProductRecordsForStore($storeId);
+            }
+        } else {
+            $this->scanProductRecordsForAllStores();
+        }
+    }
+
+    /**
+     * @return void
+     */
+    protected function scanProductRecordsForAllStores(): void
+    {
+        $storeIds = array_keys($this->storeManager->getStores());
+
+        foreach ($storeIds as $storeId) {
+            $this->scanProductRecordsForStore($storeId);
+        }
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    protected function scanProductRecordsForStore(int $storeId): void
+    {
+        $storeName = $this->storeNameFetcher->getStoreName($storeId);
+
+        if (!$this->configHelper->isIndexingEnabled($storeId)) {
+            $this->output->writeln('<info>Indexing is disabled for store ' . $storeName . '</info>');
+            return;
+        }
+
+        if (!isset($this->storeCounts[$storeId])) {
+            $this->setStoreCounts($storeId);
+        }
+
+        $this->output->writeln(' ');
+        $this->output->writeln('<info> ====== Products for store ' . $storeName . ' ====== </info>');
+        $this->output->writeln('<comment>Simple Products</comment>:  ' . $this->storeCounts[$storeId]['simple'] . ' (' . round($this->storeCounts[$storeId]['simple_percentage'], 2)  . '% of total)');
+        $this->output->writeln('<comment>Complex Products</comment>: ' . $this->storeCounts[$storeId]['complex'] . ' (' . round($this->storeCounts[$storeId]['complex_percentage'], 2) . '% of total)');
+
+        $this->output->writeln('<info> ============ </info>');
+        $this->output->writeln('<comment>Total</comment>: ' . $this->storeCounts[$storeId]['total'] . ' products');
+
+        $this->output->writeln('<info> ============ </info>');
+
+        $sample = $this->storeCounts[$storeId]['sample'];
+
+        if (count($sample) > 0) {
+            $this->output->writeln('<comment>Sample (' . count($sample) . ' products):</comment>');
+            foreach ($sample as $sku => $size) {
+                $this->output->writeln(' - ' . $size . 'B (sku: ' . $sku . ')');
+            }
+        }
+
+        $this->output->writeln('<info> ============ </info>');
+        $sizeAverage = $this->getSizeAverage($sample);
+        $this->output->writeln('<comment>Average record size</comment>         : ' . $sizeAverage . 'B');
+
+        $estimatedBatchCount = $this->getEstimatedMaxBatchCount($sizeAverage);
+        $this->output->writeln('<comment>Estimated Max batch count</comment>   : ' . $estimatedBatchCount . ' records');
+
+        $standardDeviation = $this->getStandardDeviation($sample, $sizeAverage);
+        $this->output->writeln('<comment>Standard Deviation</comment>          : ' . $standardDeviation);
+
+        $recommendedBatchCountLow = $this->getRecommendedBatchCount($sizeAverage, $standardDeviation, self::INCREASED_MARGIN);
```

Comment:
If we know the max possible size in the sample we can recommend an absolute floor that would be below the standard deviation.